src/DropdownTypeaheadReferenceSet/components/DropdownTypeaheadReferenceSet.ts

```diff
@@ -0,0 +1,137 @@
+import { Component, createElement } from "react";
+import Select , { Async } from "react-select";
+import * as classNames from "classnames";
+
+import { Alert } from "./Alert";
+import { Label } from "./Label";
+
+import "../ui/DropdownTypeaheadReferenceSet.scss";
+import "react-select/dist/react-select.css";
+
+export interface DropdownTypeaheadProps {
+    styleObject?: object;
+    labelWidth: number;
+    data?: ReferenceOption[];
+    asyncData?: any;
+    labelCaption: string;
+    loaded: boolean;
+    showLabel: boolean;
+    emptyOptionCaption: string;
+    isClearable: boolean;
+    isReadOnly: boolean;
+    selectType: "normal" | "asynchronous";
+    selectedValue: any;
+    handleOnchange?: (selectedOption: ReferenceOption | any) => void;
+    className?: string;
+    readOnlyStyle: "control" | "text";
+    labelOrientation: "horizontal" | "vertical";
+    alertMessage: string;
+}
+
+export interface ReferenceOption {
+    value?: string | boolean;
+    label?: string;
+}
+
+export interface AttributeType {
+    name: string;
+    sort: string;
+}
+
+export class DropdownTypeahead extends Component<DropdownTypeaheadProps> {
+    render() {
+        return this.renderForm();
+    }
+
+    private renderForm() {
+        if (!this.props.loaded) {
```

Comment:
or should the prob be named `loading...`
```suggestion
        if (this.props.loaded) {
```

Reply:
Done